flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/physical/batch/BatchExecSink.scala

```diff
@@ -77,11 +78,12 @@ class BatchExecSink[T](
   override protected def translateToPlanInternal(
       tableEnv: BatchTableEnvironment): StreamTransformation[Any] = {
     val resultTransformation = sink match {
-      case batchTableSink: BatchTableSink[T] =>
+      case boundedTableSink: StreamTableSink[T] =>
+        // we can insert the bounded DataStream into a StreamTableSink
         val transformation = translateToStreamTransformation(withChangeFlag = false, tableEnv)
         val boundedStream = new DataStream(tableEnv.streamEnv, transformation)
-        batchTableSink.emitBoundedStream(
-          boundedStream, tableEnv.getConfig, tableEnv.streamEnv.getConfig).getTransformation
+        boundedTableSink.emitDataStream(boundedStream)
```

Comment:
Why not just invoke `getOutputFormat` of `BoundedTableSink`?

Reply:
Because it is a `StreamTableSink`.

Reply:
According to the current design:
withChangeFlag should be false when source is BoundedTableSink
withChangeFlag should be true when source is BoundedTableSink
I think it's better to separate them and separate BoundedTableSink from DataStream.

Reply:
In batch, the `withChangeFlag` should always be false. It doesn't recognize RetractStreamTableSink or UpsertStreamTableSink. We support `StreamTableSink` in batch in order to support various streaming sinks (e.g. kafka).

Reply:
No,
If you implement a `RetractSink`(`Tuple<Boolean, Row>`), you want to reuse it in Batch, There must a true `withChangeFlag` to let conversion to convert `BaseRow` to `Tuple`, That's what `withChangeFlag` does. (Internal Blink batch support RetractSink and UpsertSink)

If you don't want support RetractSink and UpsertSink now, please throw exception when sink is RetractSink or UpsertSink.

Reply:
Sure, I will add some checks for this.